docs/specs/component.md

```diff
@@ -218,6 +168,66 @@ implement since errors are specific to the component.
 *All components* that can drop events MUST emit a `ComponentEventsDropped`
 event in accordance with the [EventsDropped event] requirements.
 
+#### SinkNetworkBytesSent
+
+*Sinks* MUST emit a `SinkNetworkBytesSent` that represents the egress of
+*raw network bytes*.
+
+- Emission
+  - MUST emit immediately after egress of raw network bytes regardless
+    if the transmission was successful or not.
+  - MUST emit *after* processing of the bytes (encryption, compression,
+    filtering, etc.)
+  - MUST NOT emit for pull-based sinks since they do not send data. For
+    example, the `prometheus_exporter` sink MUST NOT emit this event.
+- Properties
+  - `byte_size` - REQUIRED, number of raw network bytes sent after processing.
+    - SHUOLD be the closest representation possible of raw network bytes based
+      on the sink's capabilities. For example, if the sink uses a HTTP
+      client that does not provide access to the total request byte size, then
+      the sink should use the byte size of the payload/body.
+  - `url` - REQUIRED, the URL of the downstream destination.
+    - For HTTP, MUST not include the query string, only the host and path.
+      (i.e., `http(s)://<host>:<port>/<path>`)
```

Comment:
This turns out to not be a great idea for sinks like the AWS S3 sink where the object key is part of the URL.

My suggestion would be that we drop `url` entirely until someone asks for it. I think the only sink where it'd be useful to users is the `http` sink, but even then I'm not sure.